image_tools/CMakeLists.txt

```diff
@@ -45,33 +45,18 @@ if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
 
+  # get the rmw implementations ahead of time
   find_package(rmw_implementation_cmake REQUIRED)
+  get_available_rmw_implementations(rmw_implementations)
+  foreach(rmw_implementation ${rmw_implementations})
+    find_package("${rmw_implementation}" REQUIRED)
+  endforeach()
```

Comment:
I don't see where this is true (if we consider "executable" to be a compiled executable rather than an executable Pythonscript):

> we are making test executables for each rmw implementation

As far as I can see the `testing_targets` macros doesn't contain any compiled targets. It only expands a Python template and registers that Python file as a nosetest. So there is nothing which uses the `find_package` information. For the referenced tests in `system_tests` that is different - the function there actually builds executables for each rmw impl.